File: AtCoder/ABC/154/python/E_Almost_Everywhere_Zero.py
S = input()
N = len(S)
K = int(input())


def com(n, k):
    if n <= 0 or k > n:
        return 0
    if k == 0:
        return 1
    x = n
    for i in range(n - 1, n - k, -1):
        x *= i
    y = k
    for i in range(k - 1, 0, -1):
        y *= i
    return x // y


def rec(i, k, smaller):
    if i == N:
        return int(k == 0)
    if k == 0:
        return 1
    if smaller:
        return com(N - i, k) * (9 ** k)
    else:
        if S[i] == '0':
            return rec(i + 1, k, False)
        else:
            zero = rec(i + 1, k, True)
            md = rec(i + 1, k - 1, True) * (int(S[i]) - 1)
            ub = rec(i + 1, k - 1, False)
            return zero + md + ub


print(rec(0, K, False))

File: AtCoder/ABC/154/cpp/E_Almost_Everywhere_Zero.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

string S;
ll N, K;

ll com(ll n, ll k) {
  if (n <= 0 || k > n) {
    return 0;
  }
  if (k == 0) {
    return 1;
  }
  ll x = n;
  for (ll i = n - 1; i > n - k; --i) {
    x *= i;
  }
  ll y = k;
  for (ll i = k - 1; i > 0; --i) {
    y *= i;
  }
  return x / y;
}

ll p(ll x, ll n) { return (ll)pow(x, n); }

ll rec(ll i, ll k, bool smaller) {
  if (i == N) {
    return k == 0;
  }
  if (k == 0) {
    return 1;
  }
  if (smaller) {
    return com(N - i, k) * p(9, k);
  } else {
    if (S[i] == '0') {
      return rec(i + 1, k, false);
    } else {
      ll zero = rec(i + 1, k, true);
      ll md = rec(i + 1, k - 1, true) * (S[i] - '1');
      ll ub = rec(i + 1, k - 1, false);
      return zero + md + ub;
    }
  }
}

void solve() {
  N = S.size();
  printf("%lld\n", rec(0, K, false));
}

int main() {
  cin >> S >> K;
  solve();
}